### PR TITLE
JSAssignment 4 : Promises in ECMA Script

### PR DESCRIPTION
**Pull request by swatipatil-webonise**
Problem statement : 
1) Write a ES code to print 1 to 10 using promise.
2) Write a ES to do devision of two numbers.
3) Write a ES to do substraction of two numbers.
4) Write a ES to give demo of promise using delay (custom created function).

### CODE REVIEW

File: assignment4/devisionUSingPromise.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            });
+
+            promise.
+            then(function () {
+                console.log(`Devision = ${number1/number2}`);
+            }).
+            catch(function () {
+                console.log(`Can't devide by zero`);
+            });
+        </script>
```
Comment (deepakb-webonise) [inline assignment4/devisionUSingPromise.html]: The Assignment is incorrect. Create a function to add two numbers. the function should return a promise. based on promise resolve or reject, result should be shown.

File: assignment4/print1To10UsingPromise.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+        <script>
+            const TIME_TO_SLEEP = 1000;
+            async function printNumbers(endNumber) {
```
Comment (deepakb-webonise) [inline assignment4/print1To10UsingPromise.html]: Looks like the assignment is copied from the internet. async and await is introduced in es8 and we haven't covered it.

File: assignment4/print1To10UsingPromise.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+                for (let iterator = 1; iterator <= endNumber; iterator++) {
+                    console.log(await new Promise(resolver => setTimeout(resolver, TIME_TO_SLEEP, iterator)));
+                }
+            }
+            printNumbers(10).then(reject => console.log('rejected'));
```
Comment (deepakb-webonise) [inline assignment4/print1To10UsingPromise.html]: if a single function provided then it is called on successfully resolved.

File: assignment4/promiseDemo.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            then(function() {
+                console.log('Delay done');
+            }).catch(function() {
+                console.log('Delay not done');
+            })
+
+            function delay(ML) {
+                setTimeout(() => {} , ML);
+                return true;
+            }
```
Comment (deepakb-webonise) [inline assignment4/promiseDemo.html]: The assignment is incorrect. Create a delay function that will accept the delay seconds and then will return the promise which will resolve in given delay seconds.

File: assignment4/substractionUsingPromise.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+                }
+            });
+
+            promise.
+            then(function () {
+                console.log(`Substraction = ${number1-number2}`);
+            }).
+            catch(function () {
+                console.log(`Substarction is negative.`);
+            });
```
Comment (deepakb-webonise) [inline assignment4/substractionUsingPromise.html]: The Assignment is incorrect. Create a function to subtract two numbers. the function should return a promise. based on promise resolve or reject, the result should be shown.

File: assignment4/devisionUSingPromise.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+        <script>
+            function getPromise() {
+                let number1 = parseInt(document.getElementById('number1').value);
```
Comment (deepakb-webonise) [inline assignment4/devisionUSingPromise.html]: This function is tightly coupled with inputs "number1" and "number2". please remove tightly coupling.

File: assignment4/print1To10UsingPromise.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+        <script>
+            function getPromise(number) {
```
Comment (deepakb-webonise) [inline assignment4/print1To10UsingPromise.html]: give the proper and meaningful function name

File: assignment4/print1To10UsingPromise.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+                return new Promise(function(resolve , reject) {
+                     if(number <= 10){
```
Comment (deepakb-webonise) [inline assignment4/print1To10UsingPromise.html]: no need the check the number. make it generic

File: assignment4/promiseDemo.html [hunk 9]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+        <script>
+           function delay(ML) {
+               return new Promise(function (resolve, reject) {
+                   if (ML > 0) {
```
Comment (deepakb-webonise) [inline assignment4/promiseDemo.html]: no need for this condition

**Review by deepakb-webonise — APPROVED** [review]

File: assignment4/devisionUSingPromise.html [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+                        reject(`Can't devide by zero`);
+                    }
+                });
+            }
+
+            function doDevision() {
+                let number1 = parseInt(document.getElementById('number1').value);
+                let number2 = parseInt(document.getElementById('number2').value);
+                getPromiseToDoDivision(number1,number2).
+                then(function (succeed) {
```
Comment (swapnil-webonise) [inline assignment4/devisionUSingPromise.html]: Use flat arrow functions

https://eslint.org/docs/rules/prefer-arrow-callback

File: assignment4/substractionUsingPromise.html [hunk 11]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+        <script>
+            function getPromiseToDoSubstraction(number1, number2) {
```
Comment (swapnil-webonise) [inline assignment4/substractionUsingPromise.html]: No necessary to use `Promise` word in the function name, simply can be `doSubstraction`

File: assignment4/substractionUsingPromise.html [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+                    if (!isNaN(number2) && !isNaN(number1)) {
+                        resolve(`Subtraction = ${number1 - number2}`);
+                    } else {
+                        reject(`Substraction is negative.`);
+                    }
+                });
+            }
+
+            function doSubtraction() {
+                let number1 = parseInt(document.getElementById('number1').value);
```
Comment (swapnil-webonise) [inline assignment4/substractionUsingPromise.html]: Prefer `const` over `let`, As we are not modifying value futher

File: assignment4/substractionUsingPromise.html [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+                let number2 = parseInt(document.getElementById('number2').value);
```
Comment (swapnil-webonise) [inline assignment4/substractionUsingPromise.html]: Same here

File: assignment4/promiseDemo.html [hunk 14]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+        <script>
+            function delay(ML) {
```
Comment (swapnil-webonise) [inline assignment4/promiseDemo.html]: variable name should be in camelCase and descriptive

File: assignment4/devisionUSingPromise.html [hunk 15]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+        <script>
+            function getPromiseToDoDivision(number1, number2) {
+                return new Promise(function (resolve, reject) {
+                    if(number2 !== 0) {
```
Comment (swapnil-webonise) [inline assignment4/devisionUSingPromise.html]: Can be written like

`if(number2) {`

File: assignment4/devisionUSingPromise.html [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+                        resolve(`Devision = ${number1/number2}`);
```
Comment (swapnil-webonise) [inline assignment4/devisionUSingPromise.html]: Check spell

File: assignment4/devisionUSingPromise.html (hunk 10 above)
Comment (swapnil-webonise) [reply assignment4/devisionUSingPromise.html]: Check this across PR

**Review by swapnil-webonise — CHANGES_REQUESTED** [review]

**Review by swapnil-webonise — APPROVED** [review]

**Pull request merged**